### DISCUSSION TITLE
wrong firmware in ESP32S2_WROVER

### DISCUSSION THREAD

**Discussion by robhamerling** [micropython/micropython#12096] [ESP32]
When trying to update an ESP32-S2_WROVER with 1.20 esptool.py produces the message: Failed to connect to ESP32-S2: No serial data received. 
When I connect with a serial terminal via USB it seems to work fine: I can execute scripts, os.listdir() shows the files, etc. 
Then I noticed at power-on reset:
```
MicroPython v1.19.1-976-g711bac511-dirty on 2023-03-20; LOLIN_S2_MINI with ESP32-S2FN4R2
Type "help()" for more information.
>>> 
```
Apparently I loaded this ESP32 last time with the wrong firmware: for a LOLIN-S2 mini!
All efforts to let it accept new firmware failed, also an erase_flash gives the "no serial data received" message.
I could not find a solution at the Espressif troubleshooting pages. What can I do to overcome this situation?

**_False alarm: Mistake with putting the ESP32 in bootloader mode!_**

**Comment by bidrohini** [comment]
I think now you need to enter the bootloader mode and upload the correct bootloader. You can see the docs for details:
https://docs.espressif.com/projects/esptool/en/latest/esp32s2/esptool/entering-bootloader.html

**Reply by robert-hh** [reply]
@bidrohini  It's right that the device has to be in bootloader mode to upload firmware. However, the primary bootloader is located in ROM and cannot be uploaded or overwritten. It is then used to upload firmware of any kind. In the MicroPython context that is usually the MicroPython firmware, which may include a secondary bootloader, as it seems to be for the Arduino ESP32 board.

**Reply by bidrohini** [reply]
Ok, so what is the solution of the questioner's problem now?

**Reply by robert-hh** [reply]
He had written that already when I wrote my answer (see the bold sentence in the first post): use the bootloader mode for bootloading.

**Comment by robhamerling** [comment]
@bidrohini: the solution is to correctly use the right buttons on the board to get the device in bootloader mode.
